src/app.py

```diff
@@ -132,54 +133,109 @@ async def get_site_info():
     """
     site_id = request.args.get('id', type=int, )
 
-    logger.debug("User requested site detailed info", )
+    logger.debug(
+        f"User {g.authenticated_user} requested site {site_id} detailed info")
 
     async with async_session() as session:
         site = await get_site_by_id(session, site_id)
 
     return (site, 200) if site is not None else ({"error": "Site not found", }, 404)
 
+# Stamping endpoints
 
-@application.route('/api/stamp_token', methods=['GET', ], )
+
+@application.route('/api/get_stamp_token', methods=['GET', ], )
 async def stamp_token() -> Tuple[Dict[str, str], int]:
+    """
+    Parameter for getting a token to be scanned (via QR)
+    Requires user to be an employee. Token has a 30sec validity.
+
+    returns:
+        {"stamp_token" : str}
+
+    excepts:
+        401: Not logged in
+        401: not employee
+        400: Employee without assigned place
+
+    """
     async with async_session() as session:
-        result: Optional[Dict[str, Any]] = await Employees.by_id(session, g.authenticated_user)
 
-        if result is None:
+        logger.debug(
+            f"user with id {g.authenticated_user} requested a stamp token")
+
+        # Get employee info
+        employee: Optional[Dict[str, Any]] = await Employees.by_id(session, g.authenticated_user)
+
+        if employee is None:
+            logger.warning(
+                f"Non employee (id:{g.authenticated_user}) requested stamp token!")
             return {
                 'error': 'Only allowed for employees!',
+            }, 401
+
+        if employee.get('place_id') is None:
+            logger.warning(
+                f"Employee with id {g.authenticated_user} doesn't have an allocated place!")
+            return {
+                'error': 'You dont have an assigned place! Please contact an admin',
             }, 400
 
         return {
-            'token': generate_stamp_token(
+            'stamp_token': generate_stamp_token(
                 g.authenticated_user,
-                result['place_id'],
+                employee['place_id'],
             ),
         }, 200
 
 
-@application.route('/api/receive_stamp/<string:token>', methods=['POST', ], )
-async def receive_stamp(token: str) -> Tuple[Dict[str, str], int]:
+@application.route('/api/receive_stamp', methods=['POST', ], )
+async def receive_stamp() -> Tuple[Dict[str, str], int]:
+    """
+    Endpoint for receiving a stamp from a scanned token.
+
+    args:
+        'stamp_token' : str -> scanned token
+
+    returns:
+        {"message" : str} on success
+
+    excepts:
+        401: not authorized
+        400: expired/invalid token
+        400: employee trying to give himself a stamp
+        400: already have this stamp
+    """
     async with async_session() as session:
-        stamp: Stamp = finish_stamping(token, g.authenticated_user)
+
+        # Make a stamp object
+        stamp_token: str = (await request.form).get('stamp_token')
```

Comment:
Pretty sure that either returns `None` or raises an exception like that.

Reply:
Actually it works perfectly